server/src/modules/contests/contests.service.ts

```diff
@@ -404,7 +406,14 @@ export class ContestsService {
     if (isAdmin && newState === ContestState.Published) {
       this.logger.log(`Publishing contest ${contest.competitionId}...`);
 
-      if (contest.type !== ContestType.WcaComp && contest.participants < C.minCompetitorsForUnofficialCompsAndMeetups) {
+      if (contest.type === ContestType.WcaComp) {
+        const response = await fetch(`https://www.worldcubeassociation.org/api/v0/competitions/${competitionId}/results`);
+        const data = await response.json();
+        if (data && data.length === 0) {
```

Comment:
This should be `!data || data.length === 0`, because if the response data is undefined, it would just treat it as published.